jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/ValueReaderLocator.java

```diff
@@ -452,7 +452,18 @@ protected BeanReader _resolveBeanForDeser(Class<?> raw, POJODefinition beanDef)
         final Map<String, BeanPropertyReader> propMap;
         Map<String, String> aliasMapping = null;
 
-        boolean isRecord = RecordsHelpers.isRecordType(raw);
+        final Map<String, Integer> ctorParameterIndexMap = new HashMap<>();
+        final boolean isRecord = RecordsHelpers.isRecordType(raw);
+        if (isRecord) {
+            final Parameter[] parameters = constructors._recordCtor.getParameters();
+            for (int i = 0; i < parameters.length; i++) {
+                Parameter parameter = parameters[i];
+                if (!parameter.getName().isEmpty()) {
```

Comment:
Why such check? Does this happen? Isn't it an error condition?

And specifically if this happens wouldn't it just lead to an NPE when looking up the index?

Reply:
Actually, I think this is an irrelevant check, as you can not create a (record) instance variable without a name... It would lead to a NPE in the look-up part. I'd say just remove the check, as you can not test this? WDYT?

Reply:
Yes, sounds good. But as per my note above, might be this part of code can be removed altogether.